packages/apps/src/microsoft/teams/apps/options.py

```diff
@@ -5,27 +5,35 @@
 
 from dataclasses import dataclass, field
 from logging import Logger
-from typing import Any, List, Optional
+from typing import Any, List, Optional, TypedDict
 
 from microsoft.teams.common.storage import Storage
 
 from .plugins import PluginBase
 
 
-@dataclass
-class AppOptions:
+class AppOptions(TypedDict, total=False):
     """Configuration options for the Teams App."""
 
     # Authentication credentials
-    client_id: Optional[str] = None
-    client_secret: Optional[str] = None
-    tenant_id: Optional[str] = None
+    client_id: Optional[str]
+    client_secret: Optional[str]
+    tenant_id: Optional[str]
 
     # Infrastructure
-    logger: Optional[Logger] = None
-    storage: Optional[Storage[str, Any]] = None
-    plugins: List[PluginBase] = field(default_factory=list[PluginBase])
-    enable_token_validation: bool = True
+    logger: Optional[Logger]
+    storage: Optional[Storage[str, Any]]
+    plugins: List[PluginBase]
+    enable_token_validation: bool
 
     # Oauth
+    default_connection_name: str
+
+
+@dataclass
+class AppOptionsDefaults:
```

Comment:
thoughts on a factory method instead for defaults, so we don't need to maintain the dict/class?
ex - https://stackoverflow.com/a/76251992

Reply:
^ this might also help with type checking cause a `TypedDict` is not enforced at runtime

Reply:
Like it! Let me update